app/models/job_invocation.rb

```diff
@@ -193,11 +193,14 @@ def failed_hosts
   end
 
   def total_hosts_count
+    count = _('N/A')
     if targeting.resolved?
-      task&.main_action&.total_count || targeting.hosts.count
-    else
-      _('N/A')
+      count = targeting.hosts.count
+      if task&.main_action.respond_to?(:total_count)
+        count = task.main_action.total_count
+      end
```

Comment:
Could we go with something like this so targeting.hosts.count is evaluated only if action's total_count is not available?

```suggestion
      count = if task&.main_action.respond_to?(:total_count)
        task.main_action.total_count
      else
        targeting.hosts.count
      end
```